src/ServiceInsight/Shell/ServiceControlConnectionViewModel.cs

```diff
@@ -59,14 +69,20 @@ public virtual void Close()
 
         public virtual async Task Accept()
         {
+            certValidationFailed = false;
             StartWorkInProgress();
             ServiceUrl = ServiceUrl.Trim();
-            IsAddressValid = await IsValidUrl(ServiceUrl);
-            if (IsAddressValid)
+            ShowError = await IsValidUrl(ServiceUrl) == false;
+            if (!ShowError)
```

Comment:
I think it would be more readable if the showError calculation was in the if statement instead of in the await.

```CSharp
var isUrlValid = await IsValidUrl(ServiceUrl);
ShowError = !isUrlValid;
```

Reply:
refactored.